web/lua/opmserver/paginator.lua

```diff
@@ -0,0 +1,176 @@
+
+local _M = {}
+local mt = { __index = _M }
+
+local str_find = string.find
+local str_gsub = string.gsub
+local abs = math.abs
+
+function _M:new(url, total, page_size, page)
+
+    total = total or 1
+    page_size = page_size or 20
+    page = page or 1
+
+    local this = {
+        total = total,
+        page_size = page_size,
+        page = page,
+        page_count = math.ceil(total / page_size),
+        url = url
+    }
+    
+    setmetatable(this, mt)
+
+    return this
+end
+
+function _M:setUrl(page)
```

Comment:
style: we always use lowercase for lua function name, better use `set_url` here.